clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -159,7 +159,8 @@ void HLSLExternalSemaSource::defineHLSLMatrixAlias() {
                                                   SourceLocation(), ColsParam));
   TemplateParams.emplace_back(ColsParam);
 
-  const unsigned MaxMatDim = 4;
+  const unsigned MaxMatDim = SemaPtr->getLangOpts().MaxMatrixDimension;
+  ;
```

Comment:
Looks like a spurious `;`.

```suggestion
  const unsigned MaxMatDim = SemaPtr->getLangOpts().MaxMatrixDimension;
```